RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc

```diff
@@ -514,6 +513,256 @@ EcalUncalibRecHitWorkerMultiFit::run( const edm::Event & evt,
+                    
+                    // consider flagging as kOutOfTime only if above noise
+                    if (uncalibRecHit.amplitude() > pedRMSVec[0] * amplitudeThreshEE_){
+                      float outOfTimeThreshP = outOfTimeThreshG12pEE_;
+                      float outOfTimeThreshM = outOfTimeThreshG12mEE_;
+                      // determine if gain has switched away from gainId==1 (x12 gain)
+                      // and determine cuts (number of 'sigmas') to ose for kOutOfTime
+                      // >3k ADC is necessasry condition for gain switch to occur
+                      if (uncalibRecHit.amplitude() > 3000.){
+                        for (int iSample = 0; iSample < EEDataFrame::MAXSAMPLES; iSample++) {
+                          int GainId = ((EcalDataFrame)(*itdg)).sample(iSample).gainId();
+                          if (GainId!=1) {
+                            outOfTimeThreshP = outOfTimeThreshG61pEE_;
+                            outOfTimeThreshM = outOfTimeThreshG61mEE_;
+                            break;
+                          }
+                        }}
+                      float correctedTime = (crh.timeMax-5) * clockToNsConstant + itimeconst + offsetTime;
+                      float cterm         = EEtimeConstantTerm_;
+                      float sigmaped      = pedRMSVec[0];  // approx for lower gains
+                      float nterm         = EEtimeNconst_*sigmaped/uncalibRecHit.amplitude();
+                      float sigmat        = std::sqrt( nterm*nterm  + cterm*cterm   );
+                      if ( ( correctedTime > sigmat*outOfTimeThreshP )   ||
+                           ( correctedTime < -sigmat*outOfTimeThreshM) ) 
+                        {  uncalibRecHit.setFlagBit( EcalUncalibratedRecHit::kOutOfTime ); }
+                    }
+
+                } else {
+                    ratioMethod_barrel_.init( *itdg, *sampleMask_, pedVec, pedRMSVec, gainRatios );
+                    ratioMethod_barrel_.fixMGPAslew(*itdg);
+                    ratioMethod_barrel_.computeTime( EBtimeFitParameters_, EBtimeFitLimits_, EBamplitudeFitParameters_ );
+                    ratioMethod_barrel_.computeAmplitude( EBamplitudeFitParameters_);
+                    EcalUncalibRecHitRatioMethodAlgo<EBDataFrame>::CalculatedRecHit crh = ratioMethod_barrel_.getCalculatedRecHit();
+                    
+                    double theTimeCorrectionEB = timeCorrection(uncalibRecHit.amplitude(),
+                                                                timeCorrBias_->EBTimeCorrAmplitudeBins, timeCorrBias_->EBTimeCorrShiftBins);
+                    
+                    uncalibRecHit.setJitter( crh.timeMax - 5 + theTimeCorrectionEB);
+                    //uncalibRecHit.setJitterError( std::sqrt(std::pow(crh.timeError,2) + std::pow(EBtimeConstantTerm_,2)/std::pow(clockToNsConstant,2)) );
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant